packages/swapper/src/swappers/ButterSwap/swapperApi/checkTradeStatus.ts

```diff
@@ -0,0 +1,29 @@
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+
+import type { CheckTradeStatusInput, TradeStatus } from '../../../types'
+import { getBridgeInfoBySourceHash } from '../xhr'
+
+// See: https://docs.butternetwork.io/butter-swap-integration/butter-api-for-swap-data/get-swap-history-by-source-hash
+// Example endpoint: https://bs-router-v3.chainservice.io/api/queryBridgeInfoBySourceHash?sourceHash=0x...
+
+export const checkTradeStatus = async (input: CheckTradeStatusInput): Promise<TradeStatus> => {
+  const { txHash } = input
+  try {
+    const info = await getBridgeInfoBySourceHash(txHash)
+    if (!info) {
+      console.debug('ButterSwap checkTradeStatus: no info', { txHash })
+      return { status: TxStatus.Unknown, buyTxHash: undefined, message: 'No data from ButterSwap' }
```

Comment:
I don't think we want this. Let's keep to the default `undefined` as we do everywhere so we can keep default messaging:

<img width="529" alt="image" src="https://github.com/user-attachments/assets/482d33fb-1ecb-4d60-967b-71fdd1a70611" />

Reply:
https://github.com/shapeshift/web/pull/9799/commits/3969c14dbcf456aabed32705af313dbbf7bf5e56